### ISSUE TITLE
Show an indicator when the creative mode is active

### ISSUE THREAD

**Issue by kneekoo** [minetest/minetest_game#2890]
Full details in #2838 - about the unexpected game mode switch **and** having no visual clue about the creative mode being active.

In the long run, [Minetest should support multiple game modes](https://github.com/minetest/minetest/issues/11457), with the player having the ability to explicitly switch to them, without the game having to guess which one is supposed to become active when you have privileges for all of them.

**Comment by sfan5** [comment]
The main point here seems to be "why is creative mode enabled?" and not "is creative mode enabled?" (the creative menu is impossible to miss as an indicator).
Do you think an indicator text in the inventory that activates when `creative` is granted would help?

**Comment by kneekoo** [comment]
No, the main point is knowing if the creating mode is enabled, because you can use `/grantme all` while exploring/mining/etc and not use the inventory at all for a while, so you wouldn't know why you can't gather more stuff (incrementing counters).

The HUD/hotbar could have a side icon to make it clear the game is in creative mode.

**Comment by Montandalar** [comment]
There are icons for fast, fly, noclip from the Android version. Add a new one for creative and display them above the health bar? They would have on/off icons variants so you can tell whether you have something on or not.

**Comment by appgurueu** [comment]
> There are icons for fast, fly, noclip from the Android version. Add a new one for creative and display them above the health bar? They would have on/off icons variants so you can tell whether you have something on or not.

To me this just seems like polluting the HUD with an element that doesn't really convey useful information. This probably belongs in the status message or the debug info. But a permanently on-screen indicator would quickly become obnoxious.

**Comment by kneekoo** [comment]
Minecraft does it quite subtle. In survival, the player has the experience bar displayed atop the hotbar. In creative, there's no XP bar. So there's no special indicator for creative mode, but you can infer it from the HUD's look. So there are multiple ways to handle this.

**Comment by rollerozxa** [comment]
Minecraft also couples damage with creative mode though (which minetest doesn't, it is separated into two different settings)

**Comment by Montandalar** [comment]
Would it be possible to dynamically change the hudbar texture so that it's different between creative and survival players? I'm not sure what that would look like in graphic design, but I'm sure someone can think of something appropriate (and hopefully not just an arbitrary colour coding system).
